clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -256,14 +271,176 @@ private void process(final UnsubscribeEvent event) {
         }
     }
 
-    private void process(final ResetPositionsEvent event) {
-        CompletableFuture<Void> future = requestManagers.offsetsRequestManager.resetPositionsIfNeeded();
-        future.whenComplete(complete(event.future()));
+    /**
+     *
+     * Fetch committed offsets and use them to update positions in the subscription state. If no
+     * committed offsets available, fetch offsets from the leader.
+     */
+    private void process(final UpdateFetchPositionsEvent updateFetchPositionsEvent) {
+        try {
+            // The event could be completed in the app thread before it got to be
+            // processed in the background (ex. interrupted)
+            if (updateFetchPositionsEvent.future().isCompletedExceptionally()) {
+                log.debug("UpdateFetchPositions event {} was completed exceptionally before it " +
+                    "got time to be processed.", updateFetchPositionsEvent);
+                return;
+            }
```

Comment:
Similar to the code in `AsyncKafkaConsumer`, isn't this case true for any of the `CompletableApplicationEvent` subclasses? If that's the case, we should probably handle it in a consistent way at a higher layer 🤔

Reply:
Agreed, comment above but basically moved to the addAndGet as a general case